src/auth/jwt.strategy.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, UnauthorizedException } from '@nestjs/common';
+import { PassportStrategy } from '@nestjs/passport';
+import { ExtractJwt, Strategy } from 'passport-jwt';
+import { Env } from 'src/env/index';
+import { JwtPayload } from './jwt-interface';
+import { RedisService } from '@liaoliaots/nestjs-redis';
+
+@Injectable()
+export class JwtStrategy extends PassportStrategy(Strategy, 'jwt') {
+  constructor(
+    private readonly redisService: RedisService,
+  ) {
+    super({
+      secretOrKey: Env.JWT.AUTH_TOKEN,
+      jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
+    });
+  }
+
+  async validate(payload: JwtPayload) {
+    const { userId } = payload;
+    const client = this.redisService.getClient();
+    const jwtToken = await client.get(`jwt:${userId}`);
```

Comment:
PullProofAI

Consider adding error handling for the asynchronous operation `client.get` to handle potential promise rejections.
```typescript
try {
    const jwtToken = await client.get(`jwt:${userId}`);
} catch (error) {
    // Handle error, e.g., log or throw a custom exception
}
```


<!-- This is an auto-generated comment by PullProofAI -->